### ISSUE TITLE
EDITOR: Assertion failure when trying to use the transparent material

### ISSUE THREAD

**Issue by christianclavet** [BearishSun/BansheeEngine#113]
Build date: (16 december 2017)
How to replicate the problem:
Create a new material, and change the material type from standard to transparent.

Result: Crash with assertion failure:
Here is the output from the MSVC debugger output (using a debug build):
Debug Assertion Failed!

```
Program: C:\WINDOWS\SYSTEM32\MSVCP140D.dll
File: c:\program files (x86)\microsoft visual studio 14.0\vc\include\vector
Line: 1234

Expression: vector subscript out of range

For information on how your program can cause an assertion
failure, see the Visual C++ documentation on asserts.

(Press Retry to debug the application)
BansheeEditorExec.exe a déclenché un point d'arrêt. 
```
Did you change the way you call the parameters of the transparent shader?

**Comment by BearishSun** [comment]
This should be resolved with c7e198eda144cdac5a2971c28b1fa7d5fbe7c43c.

**Comment by christianclavet** [comment]
This fixed the issue, had transparent material. BUT when I saved the scene, quit the application and reloaded the application and scene, the transparent material was no more.

Checked the material, and it was back to standard, selecting transparent made it crash again. Seem not saving the project, had this material back to "default"

So I deleted the material, created a new one with the same settings, saved the scene AND project.
I exited the application again, reloaded everything and it was ok. Seem that the material is not saved in the project when we set it, and would need to save manually, and trying to set it after being reloaded (from a full exit and reload), will make the application crash.

**Comment by BearishSun** [comment]
The project should be automatically saved if you close it normally (hitting the X, or File/Exit in the menu). If closed abruptly (crash, or from Visual Studio by stopping debugging) the project will not be saved, and the material will indeed revert back to its original state. I've tried closing the application normally and the project was saved for me, and the material was saved as Transparent.

I've also tried to reproduce the crash, but was unable to. I created a new standard material, switched it to Transparent, forcibly closed the application, reloaded and the material was back to Standard. Switching to Transparent or setting textures worked without a crash.

There might be some steps I'm missing?
